compass/landice/tests/slm/circ_icesheet/setup_mesh.py

```diff
@@ -0,0 +1,409 @@
+        smb = -1 * ((10000.0) * rhoi / (365.0 * 24.0 * 60.0 * 60.0))
+        print(f'prescribed smb is {smb} kg/m2/s')
+        # for loop through time
+        for t in range(len(t_array)):
+            # calculate a new radius at each time
+            Rt = r0 + (drdt * (t + 1))
+            if (Rt < 0):
+                Rt = 0.0
+                sfcMassBal[t, :] = 0.0
+                print(f'At time {start_year+dt_year*t}, \
+                      new radius will be {Rt} km')
+            else:
+                indx = np.where(r >= Rt)[0]
+                sfcMassBal[t, indx] = smb
+                print(f'At time {start_year+dt_year*t}, \
+                      new radius will be {Rt/1000} km')
+        smbfile.variables['sfcMassBal'][:, :] = sfcMassBal[:, :]
+
+    # add xtime variable
+    smbfile.createDimension('StrLen', 64)
+    xtime = smbfile.createVariable('xtime', 'S1', ('Time', 'StrLen'))
+
+    # initialize SMB value everywhere in the mesh
+    xtime_str = []
+    for t_index in range(len(t_array)):
+        yr = start_year + (t_index * dt_year)
+        xtime_str = f'{int(yr)}-01-01_00:00:00'.ljust(64)
+        xtime_char = netCDF4.stringtochar(np.array([xtime_str], 'S64'),
+                                          encoding='utf-8')
+        xtime[t_index, :] = xtime_char
+
+    smbfile.close()
+
+
+def _build_mapping_files(config, logger, res, nglv, mali_mesh_file):
+    """
+    Build a mapping file if it does not exist.
+    Mapping file is then used to remap the ismip6 source file in polarstero
+    coordinate to unstructured mali mesh
+
+    Parameters
+    ----------
+    config : compass.config.CompassConfigParser
+        Configuration options for a ismip6 forcing test case
+
+    logger : logging.Logger
+        A logger for output from the step
+
+    name : str
+        Name of the step
+
+    res : str
+        Resolution of MALI mesh
+
+    nglv : str
+            Number of Gauss-Legendre nodes in latitude in the SLM
+
+    mali_mesh_file : str
+        The MALI mesh file
+    """
+
+    section = config['slm']
+    slm_res = section.get('slm_res')
+    slm_nglv = int(nglv)
+    method_mali_to_slm = section.get('mapping_method_mali_to_slm')
+    method_slm_to_mali = section.get('mapping_method_slm_to_mali')
+
+    mali_scripfile = f'mali{int(res)}km_scripfile.nc'
+    slm_scripfile = f'slm{slm_res}_nglv{slm_nglv}scripfile.nc'
+
+    # create slm scripfile
+    logger.info(f'creating scripfile for the SH-degree {slm_res} '
+                f'resolution and {slm_nglv} nglv points for the SLM grid')
+    args = ['ncremap',
+            '-g', slm_scripfile,
+            '-G',
+            f'latlon={slm_nglv},{2*int(slm_nglv)}#lat_typ=gss#lat_drc=n2s']
+
+    check_call(args, logger=logger)
+
+    # adjust the lat-lon values
+    args = ['set_lat_lon_fields_in_planar_grid.py',
+            '--file', mali_mesh_file,
+            '--proj', 'ais-bedmap2-sphere']
```

Comment:
Flagging again that we need to add this in an MPAS-Tools PR.  (Already mentioned in #749)